src/js/jsx/search/SearchBar.jsx

```diff
@@ -102,12 +108,60 @@ define(function (require, exports, module) {
             };
         },
 
-        componentDidMount: function () {
+        /**
+         * Explicitly focus the text input of the search datalist when the window is activated.
+         *
+         * @private
+         * @param {{becameActive: boolean}} event
+         */
+        _handleActivationChanged: function (event) {
+            if (this.state.ready && event && event.becameActive) {
+                this.refs.datalist.focus();
+            }
+        },
+
+        /**
+         * Initialize search providers for the given searchID.
+         *
+         * @private
+         */
+        _initSearchProviders: function () {
             var searchStore = this.getFlux().store("search");
             searchStore._updateSearchItems(this.props.searchID);
         },
 
+        componentWillMount: function () {
+            this.getFlux().actions.documents.initializeDocumentsThrottled();
+        },
+
+        componentDidMount: function () {
+            if (this.state.ready) {
+                // In case all documents are already initialized
+                this._initSearchProviders();
+            }
+
+            os.addListener("activationChanged", this._handleActivationChanged);
+        },
+
+        componentWillUnmount: function () {
+            os.removeListener("activationChanged", this._handleActivationChanged);
+        },
+
         componentDidUpdate: function (prevProps, prevState) {
+            if (!prevState.ready && this.state.ready) {
+                // In case some documents were not initialized, but now are
+                this._initSearchProviders();
+            } else if (prevState.ready && !this.state.ready) {
+                // In case there are newly uninitiazlied documents
+                this.getFlux().actions.documents.initializeDocumentsThrottled();
```

Comment:
we do not seem to be doing anything with the promise returned from here. should we be waiting for it to resolve?